### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by BarryThePenguin**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
-/node_modules
-/lib
+node_modules
```
Comment (alex-e-leon) [inline .gitignore:1]: Curious as to why you prefer no leading prefix for all folders?  Note that without the leading prefix `/src/lib` would be ignored (however node_modules probably works better without them, you WOULD want to ignore `/src/node_modules` I think).

Doesn't really matter in this case - they both work the same here.

File: package.json [hunk 2]
```diff
@@ -7,33 +7,42 @@
   "repository": "alex-e-leon/johnny-dependency",
   "author": "alex leon <alex@leon.st>",
   "license": "MIT",
+  "files": [
+    "lib"
```
Comment (alex-e-leon) [inline package.json:11]: 👍

File: src/format-dependencies.js [hunk 3]
```diff
@@ -2,7 +2,8 @@
 
 import type { PackageWithDeps, PackageAtVersion, PackageVersionMap } from './types';
 
-function resolveDependencies({ dependencies, resolvedDependencies }: {
+function resolveDependencies({
+  dependencies, resolvedDependencies }: {
```
Comment (alex-e-leon) [inline src/format-dependencies.js:6]: Is this from prettier? I'm confused as to why you'd break an object onto two lines? Feel like the original was cleaner.

File: src/index.js [hunk 4]
```diff
@@ -1,24 +1,22 @@
...
-      )
-    ))
+    getLatestDependencies(_.uniq(result.dependencies.map(npmPackage => npmPackage.name)))
+      .then(latestVersions => (
+        formatGraphWithoutCircularDeps(
+          { name: rootPackage.name, version: result.resolvedDependencies[`${rootPackage.name}@${rootPackage.version}`] },
+          resolveDependencies(result),
+          latestVersions,
+        )
+      ))
```
Comment (alex-e-leon) [inline src/index.js:19]: Again is this prettier? I actually don't really like most of the formatting suggestions it's made - I think the original is much easier to read.

File: src/types.js [hunk 5]
```diff
@@ -5,6 +5,11 @@ export type PackageAtVersion = {
   version: string
 };
 
+export type Node = PackageAtVersion & {
```
Comment (alex-e-leon) [inline src/types.js:8]: Doesn't look like this type is being used?

File: test/index.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// @flow
+
+import test from 'ava';
+import packageJson from 'print-tree/package.json';
+import jd from '../src/index';
+
+test('should retrive a list of dependencies from npm', async (t) => {
+  t.snapshot(await jd({
```
Comment (alex-e-leon) [inline test/index.js:8]: Oh man - these snapshots are so nice for an easy easy test : )

File: package.json [hunk 7]
```diff
@@ -7,33 +7,42 @@
...
-    "start": "yarn run build && ./lib/cli/index.js",
-    "prepublish": "yarn run build",
-    "lint": "./node_modules/.bin/eslint src/",
-    "test": "yarn run flow && yarn run lint"
+    "start": "yarn build && ./lib/cli/index.js",
+    "prepare": "yarn build",
+    "lint": "eslint src test",
+    "test": "flow && yarn lint && nyc ava",
+    "test:watch": "ava --watch",
+    "coverage": "nyc report --reporter=text-lcov > coverage.lcov && codecov"
```
Comment (alex-e-leon) [inline package.json:20]: Is there really a point to including coverage if we're just doing snapshot testing? 😛

**Review by alex-e-leon — APPROVED** [review]
LGTM - I still don't like some of the formatting changes - but I won't let minor code-style issues get in the way of some good tests

File: package.json (hunk 7 above)
Comment (BarryThePenguin) [reply package.json:20]: Snapshot testing doesn't imply 100% coverage? It's just an easy way of writing assertions

File: .gitignore (hunk 1 above)
Comment (BarryThePenguin) [reply .gitignore:1]: It doesn't really bother me

File: src/format-dependencies.js (hunk 3 above)
Comment (BarryThePenguin) [reply src/format-dependencies.js:6]: That's `eslint --fix`

You could add prettier if you wanted to though

File: src/types.js (hunk 5 above)
Comment (BarryThePenguin) [reply src/types.js:8]: I think I have another PR ready that does use it

Comment (alex-e-leon) [reply src/types.js:8]: 👍

File: src/format-dependencies.js (hunk 3 above)
Comment (alex-e-leon) [reply src/format-dependencies.js:6]: ewww

File: package.json (hunk 7 above)
Comment (alex-e-leon) [reply package.json:20]: True

**Pull request merged**